treeherder/services/taskcluster.py

```diff
@@ -161,3 +182,61 @@ def _task_in_context(cls, context: List[dict], task_tags: dict) -> bool:
             all(tag in task_tags and task_tags[tag] == tag_set[tag] for tag in tag_set.keys())
             for tag_set in context
         )
+
+
+class TaskclusterModelProxy(TaskclusterModel):
+    """
+    TODO: remove this class when backfill tool' soft launch is complete
+    """
+
+    def trigger_action(self, action, task_id, decision_task_id, input, root_url=None) -> str:
+        hash_suffix = self.__hash(task_id)
+        return f'fake-backfill-task-id-for-{task_id}-{hash_suffix}'
+
+    def __hash(self, task_id) -> str:
+        now = str(datetime.now())
+        return hashlib.sha256(f'{now}-{task_id}').hexdigest()
+
+
+class Notify(ABC):
+    @abstractmethod
+    def email(self, *args, **kwargs):
+        pass  # pragma: no cover
+
+
+class NotifyAdapter(Notify):
+    def __init__(self, options=None, session=None):
+        self._notify_adaptee = taskcluster.Notify(options, session)
+
+    def email(self, *args, **kwargs):
+        return self._notify_adaptee.email(*args, **kwargs)
+
+
+class NotifyNullObject(Notify):
+    def email(self, *args, **kwargs):
+        logger.debug(f"Faking sending of email `{args}`")
+
+
+def notify_client_factory(
```

Comment:
I like the idea of bringing this logic in a factory instead of having it all over the code